benchmarking/benchmark.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import pathlib
+import subprocess
+import time
+
+import openmc
+from .plot import plot
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-l', '--benchmarks', type=str,
+                        default='benchmarks/lists/pst-short',
+                        help='List of benchmarks to run.')
+    parser.add_argument('-c', '--cross-sections', type=str,
+                        default=os.getenv('OPENMC_CROSS_SECTIONS'),
+                        help='Cross sections XML file.')
+    parser.add_argument('-p', '--particles', type=int, default=10000,
+                        help='Number of source particles.')
+    parser.add_argument('-b', '--batches', type=int, default=150,
+                        help='Number of batches.')
+    parser.add_argument('-i', '--inactive', type=int, default=50,
+                        help='Number of inactive batches.')
+    parser.add_argument('-m', '--max-batches', type=int, default=10000,
+                        help='Maximum number of batches.')
+    parser.add_argument('-t', '--threshold', type=float, default=0.0001,
+                        help='Value of the standard deviation trigger on '
+                        'eigenvalue.')
+    args = parser.parse_args()
+
+    current_time = time.localtime()
+    timestamp = time.strftime("%Y-%m-%d-%H%M%S", current_time)
+
+    # Create directory and set filename for results
+    cwd = pathlib.Path(__file__).parent
+    os.makedirs(cwd / 'results', exist_ok=True)
+    outfile = cwd / f'results/{timestamp}.csv'
+
+    # Get a copy of the benchmarks repository
+    if not (cwd / 'benchmarks').is_dir():
+        repo = 'https://github.com/mit-crpg/benchmarks.git'
+        subprocess.Popen(['git', 'clone', repo], cwd=cwd)
+
+    # Get the list of benchmarks to run
+    if not pathlib.Path(args.benchmarks).is_file():
+        msg = 'Could not locate the benchmark list {}.'.format(args.benchmarks)
+        raise ValueError(msg)
+    with open(args.benchmarks) as f:
+        benchmarks = f.read().split()
+
+    # Prepare and run benchmarks
+    for benchmark in benchmarks:
```

Comment:
At the beginning of this block, I would run `openmc.reset_auto_ids` to avoid warnings about materials with the same ID